File: src/Organization.Api/Domain/Companies/Company.cs
using StarterKit.Shared.Entities;

namespace StarterKit.Organization.Api.Domain.Companies;

public class Company : AuditableEntity
{
    public string Name { get; set; } = null!;

    public string Code { get; set; } = null!;

    public string? TaxCode { get; set; }

    public string? Address { get; set; }

    public string? Phone { get; set; }

    public string? Email { get; set; }

    public string? Website { get; set; }

    public string? Description { get; set; }

    public OrganizationStatus Status { get; set; } = OrganizationStatus.Active;
}

File: clients/admin/src/features/companies/types/company.ts
/** Mirrors `Organization.Contracts/Common/OrganizationStatus.cs` — the backend serializes this enum by name, not by number (same convention as `NotificationStatus`). */
export enum OrganizationStatus {
  Active = "Active",
  Inactive = "Inactive",
}

export interface CompanyDto {
  id: string;
  name: string;
  code: string;
  taxCode?: string | null;
  address?: string | null;
  phone?: string | null;
  email?: string | null;
  website?: string | null;
  description?: string | null;
  status: OrganizationStatus;
}

export interface CreateCompanyRequest {
  name: string;
  code: string;
  taxCode?: string;
  address?: string;
  phone?: string;
  email?: string;
  website?: string;
  description?: string;
}

export interface SearchCompaniesParams {
  searchValue?: string;
  pageNumber?: number;
  pageSize?: number;
  status?: OrganizationStatus;
}
